clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeployCloudFormationAtomicOperation.java

```diff
@@ -74,13 +75,57 @@ public Map operate(List priorOutputs) {
         .withKey(entry.getKey())
         .withValue(entry.getValue()))
       .collect(Collectors.toList());
+
+    String stackId = "";
+    StackStatus stackStatus = StackStatus.fromValue(getStack(amazonCloudFormation).getStackStatus());
     try {
-      String stackId = createStack(amazonCloudFormation, template, parameters, tags, description.getCapabilities());
-      return Collections.singletonMap("stackId", stackId);
-    } catch (AlreadyExistsException e) {
-      String stackId = updateStack(amazonCloudFormation, template, parameters, tags, description.getCapabilities());
-      return Collections.singletonMap("stackId", stackId);
+      switch (stackStatus) {
```

Comment:
switch statemens are not very readable when there are lots of cases imho. Given there are only two paths can we simplify it with if/else? like:
if (stackStatus == CREATE_FAILED || stackStatus == ROLLBACK_COMPLETE) {
  delete stack
  create stack
} else {
  update stack (if it throws an exception, it will already fail the task and orca will get the reason)
}